pkg/spireentry/entries.go

```diff
@@ -37,6 +37,9 @@ func renderPodEntry(spec *spirev1alpha1.ParsedClusterSPIFFEIDSpec, node *corev1.
 		{Type: "k8s", Value: fmt.Sprintf("pod-uid:%s", pod.UID)},
 	}
 	parentID, err := spiffeid.FromPathf(trustDomain, "/spire/agent/k8s_psat/%s/%s", clusterName, node.UID)
+	if spec.ParentID.String() != "" {
```

Comment:
The idea here was to use the parent ID from the entry if it exists, otherwise use the generated one above. I'm not sure if that's the correct logic. I also suspect we need more changes to be able to create the entries with the parent ID in the first place.